src/main/java/book/store/service/impl/BookServiceImpl.java

```diff
@@ -36,4 +40,31 @@ public List<BookDto> getAll() {
                 .map(bookMapper::toDto)
                 .collect(Collectors.toList());
     }
+
+    @Override
+    public boolean updateBookById(Long id, CreateBookRequestDto updateDto) {
+        Book book = bookMapper.toModel(updateDto);
+        int udapted = bookRepository
+                .updateBookByIdAndTitleAndAuthorAndIsbnAndPriceAndDescriptionAndCoverImage(id,
+                        book.getTitle(),
+                        book.getAuthor(),
+                        book.getIsbn(),
+                        book.getPrice(),
+                        book.getDescription(),
+                        book.getCoverImage());
+        return udapted > 0;

```

Comment:
Just call .save on the updated book model - it will overwrite existing with matching id. Handle the case when id not found